example/CMakeLists.txt

```diff
@@ -33,6 +33,8 @@ if(RANGES_BUILD_CALENDAR_EXAMPLE)
         message(STATUS "boost: ${Boost_LIBRARY_DIRS}")
         target_compile_definitions(calendar PRIVATE BOOST_NO_AUTO_PTR)
         target_compile_options(calendar PRIVATE -std=gnu++14)
-        set_target_properties(calendar PROPERTIES LINK_FLAGS "-L${Boost_LIBRARY_DIRS}")
+        if (${Boost_LIBRARY_DIRS})
+          set_target_properties(calendar PROPERTIES LINK_FLAGS "-L${Boost_LIBRARY_DIRS}")
+        endif()
```

Comment:
How does this work? The `calendar` target needs to link to the Boost.DateTime and Boost.ProgramOptions libraries. Does it even make sense to build the `calendar` target if those libraries aren't there?

Reply:
This checks if `Boost_LIBRARY_DIRS` is set. I guess the local installation of @xuning97 doesn't include a CMake package.

Reply:
I still don't get how `Boost_FOUND` can be true but `Boost_LIBRARY_DIRS` is empty.

Reply:
Maybe if it uses [Boost CMake](https://cmake.org/cmake/help/latest/module/FindBoost.html#boost-cmake), assuming it doesn't set it.

Reply:
Searching for it in its organization turns no matches: https://github.com/search?q=org%3Aboost-cmake+Boost_LIBRARY_DIRS&unscoped_q=Boost_LIBRARY_DIRS.

Reply:
It is set by FindBoost.cmake, which is a standard part of every CMake install: https://github.com/Kitware/CMake/blob/master/Modules/FindBoost.cmake

Reply:
As stated in the link above for Boost CMake, the [logic](https://github.com/Kitware/CMake/blob/master/Modules/FindBoost.cmake#L426) does use the installed Boost, which does not set `Boost_LIBRARY_DIRS`.

Reply:
There is no Boost_LIBRARY_DIRS in the cmake file shipped with boost sources.
https://github.com/boostorg/boost_install/blob/develop/BoostConfig.cmake